Article about custom file downloader has incorrect method call

[Letting The User Download A File](https://github.com/vaadin/framework/blob/master/documentation/articles/LettingTheUserDownloadAFile.asciidoc) article's example _Lazily determine the content and the name of the file_ calls `getResource()` method without session lock which triggers the following error when assertions are enabled `java.lang.AssertionError: Session must be locked when getState() is called`. Perhaps the example should be revised to provide a more valid solution.

--- a/documentation/articles/LettingTheUserDownloadAFile.asciidoc
+++ b/documentation/articles/LettingTheUserDownloadAFile.asciidoc
@@ -104,8 +104,11 @@ public class OnDemandFileDownloader extends FileDownloader {
     return super.handleConnectorRequest(request, response, path);
   }
 
-  private StreamResource getResource () {
-    return (StreamResource) this.getResource("dl");
+  private StreamResource getResource() {
+    this.getSession().lock();
+    StreamResource result = (StreamResource) this.getResource("dl");
+    this.getSession().unlock();
+    return result;
   }
 }
 ....

--- a/documentation/articles/LettingTheUserDownloadAFile.asciidoc
+++ b/documentation/articles/LettingTheUserDownloadAFile.asciidoc
@@ -105,9 +105,13 @@ public class OnDemandFileDownloader extends FileDownloader {
   }
 
   private StreamResource getResource() {
+    StreamResource result = null;
     this.getSession().lock();
-    StreamResource result = (StreamResource) this.getResource("dl");
-    this.getSession().unlock();
+    try {
+      result = (StreamResource) this.getResource("dl");
+    } finally {
+      this.getSession().unlock();
+    }
     return result;
   }
 }

